lib/horde/registry.ex

```diff
@@ -36,23 +36,27 @@ defmodule Horde.Registry do
 
   @callback init(options :: Keyword.t()) :: {:ok, options :: Keyword.t()}
 
-  defmacro __using__(_opts) do
-    quote do
+  defmacro __using__(opts) do
+    quote location: :keep, bind_quoted: [opts: opts] do
       @behaviour Horde.Registry
 
       def child_spec(options) do
         options = Keyword.put_new(options, :id, __MODULE__)
 
-        %{
+        default = %{
           id: Keyword.get(options, :id, __MODULE__),
           start: {__MODULE__, :start_link, [options]},
           type: :supervisor
         }
+
+        Supervisor.child_spec(default, unquote(Macro.escape(opts)))
```

Comment:
As inspired by: https://github.com/elixir-lang/elixir/blob/c76c5d6d188bbb787d6afefab74a31080695d1f2/lib/elixir/lib/gen_server.ex#L702

Personal note: I do not understand why: `unquote(Macro.escape(opts))` as opposed to simply doing: `unquote(opts)`. The former makes it difficult to override the key: `start` MFA

Reply:
I'm also not sure, if it works without this, then I think that's also fine.